scrapinghub/client/items.py

```diff
@@ -59,3 +59,30 @@ def _modify_iter_params(self, params):
         if offset:
             params['start'] = '{}/{}'.format(self.key, offset)
         return params
+
+    def iter_by_chunks(self, chunksize=10000, *args, **kwargs):
+        """An alternative for reading and processing items by returning a
+        generator of item chunks.
+
+        This is a convenient method for cases when processing a large amount of
+        items from a job isn't ideal in one go due to the large memory needed.
+        Instead, this allows you to process it chunk by chunk.
+
+        You can improve I/O overheads by increasing the chunk value but that
+        would also increase the memory consumption.
+
+        :return: an iterator over a list of elements.
```

Comment:
```suggestion
        :return: an iterator over items, yielding lists of items.
```
If we use typing, it would be more obvious.